gpkit/nomials/data.py

```diff
@@ -165,10 +165,11 @@ def simplify_exps_and_cs(exps, cs, return_map=False):
         if return_map:
             expmap[exp][i] = cs[i]
 
-    if len(matches) > 1:
-        zeroed_terms = (exp for exp, c in matches.items() if mag(c) == 0)
-        for exp in zeroed_terms:
-            del matches[exp]
+    zeroed_terms = (exp for exp, c in matches.items() if mag(c) == 0)
+    for exp in zeroed_terms:
+        del matches[exp]
+    if not matches:  # it's a zero monomial
+        matches[HashVector()] = 0
```

Comment:
interesting, why do we need to set `matches[HashVector()] = 0` here? I would have thought we just let `exps_ = (),` and `cs = []` (and therefore `mmap = []` if returned) in that case.